### ISSUE TITLE
vite plugins are bundled and served to the client

### ISSUE THREAD

**Issue by danprince** [tajo/ladle#143]
Ladle breaks when using `vite-plugin-svgr`, which imports `vite` itself as a runtime dependency. Both packages are bundled and served to the client.

Here's a [StackBlitz project](https://stackblitz.com/edit/ladle-fhglcv) with the errors.

The immediate issue is that the global replacement turns a `process.env.NODE_ENV = "development"` into `"development" = "development"` which browsers won't parse.

The larger problem is that Vite plugins (and their dependencies) probably shouldn't make it into the browser environment.

I tried using adding the plugin to a separate `vite.config.js` at the project root, but Ladle doesn't seem to be picking it up (the transforms aren't applied).

There would be some appeal in including `vite-plugin-svgr` for [out-of-the-box compatibility with Create React App](https://create-react-app.dev/docs/adding-images-fonts-and-files/#adding-svgs), but totally understandable if that's a non-goal.

I would have thought this same problem would exist for many Vite plugins, so having a way to specify them without them ending up the client is probably the better long term solution here.

Happy to send a PR if there's a clear route forwards here.

**Comment by danprince** [comment]
Figured out a temporary workaround by aliasing the plugin to a safe file with Vite.

```ts
// .ladle/config.mjs
import svgr from "vite-plugin-svgr";

export default {
  vitePlugins: [svgr()],
  resolve: {
    alias: {
      // The actual vite-plugin-svgr (in node_modules) is used at compile time
      // by node to transform svgs, but because Ladle serves the config file at
      // runtime, we need to prevent Vite from bundling it. This alias is a quick
      // hack to point towards a no-op module that can be safely served at runtime.
      // See https://github.com/tajo/ladle/issues/143 for more info.
      "vite-plugin-svgr": "./vite-plugin-svgr.js"
    },
  }
}
```

And here's the plugin stub.

```ts
// .ladle/vite-plugin-svgr.js
export default () => {};
```

Not great, but unblocks me again until I need another Vite plugin 😆

**Comment by whitestripe42** [comment]
Above workaround is nice, but avoid polluting the codebase with empty files, and less maintenance:

```
const isNode = typeof process !== 'undefined';

async function importPlugin(lib) {
  if (isNode) {
    const { default: plugin } = await import(/* @vite-ignore */ lib);
    return plugin;
  }
  return () => void 0;
}

const checker = await importPlugin('vite-plugin-checker');

...

export default {
  vitePlugins: [
    checker({
      eslint: isNode && {
        lintCommand: `eslint ${process.cwd()}/src/**/*.{js,jsx,ts,tsx}`,
      },
    }),
  ],

}
```

**Comment by tajo** [comment]
This will be addressed with v1 and supporting `vite.config.js` by default.
